include/zenoh-pico/api/primitives.h

```diff
@@ -216,594 +220,472 @@ _Bool z_keyexpr_intersects(const z_loaned_keyexpr_t *l, const z_loaned_keyexpr_t
- * To check if ``val`` is still valid, you may use ``z_scouting_config_check(&val)`` or ``z_check(val)`` if your
- * compiler supports ``_Generic``, which will return ``true`` if ``val`` is valid, or ``false`` otherwise.
- *
- * Returns:
- *   Returns a new, zenoh-allocated, default scouting configuration.
+ * Parameters:
+ *   sc: Pointer to an uninitialized :c:type:`z_owned_scouting_config_t`.
  */
 void z_scouting_config_default(z_owned_scouting_config_t *sc);
 
 /**
- * Return a new, zenoh-allocated, scouting configuration extracted from a :c:type:`z_owned_config_t`.
- * It consists in a default set of properties for scouting configuration.
- *
- * Like most ``z_owned_X_t`` types, you may obtain an instance of :c:type:`z_owned_scouting_config_t` by loaning it
- * using
- * ``z_scouting_config_loan(&val)``. The ``z_loan(val)`` macro, available if your compiler supports C11's ``_Generic``,
- * is equivalent to writing ``z_config_loan(&val)``.
- *
- * Like all ``z_owned_X_t``, an instance will be destroyed by any function which takes a mutable pointer to said
- * instance, as this implies the instance's inners were moved. To make this fact more obvious when reading your code,
- * consider using ``z_move(val)`` instead of ``&val`` as the argument. After a ``z_move``, ``val`` will still exist, but
- * will no longer be valid. The destructors are double-drop-safe, but other functions will still trust that your ``val``
- * is valid.
- *
- * To check if ``val`` is still valid, you may use ``z_scouting_config_check(&val)`` or ``z_check(val)`` if your
- * compiler supports ``_Generic``, which will return ``true`` if ``val`` is valid, or ``false`` otherwise.
+ * Build a new :c:type:`z_owned_scouting_config_t` with values from a :c:type:`z_loaned_config_t`.
  *
  * Parameters:
- *   config: A loaned instance of :c:type:`z_owned_config_t`.
+ *   config: Pointer to a :c:type:`z_owned_config_t` to get the values from.
+ *   sc: Pointer to an uninitialized :c:type:`z_owned_scouting_config_t`.
  *
- * Returns:
- *   Returns a new, zenoh-allocated, default scouting configuration.
+ * Return:
+ *   ``0`` if build successful, ``negative value`` otherwise.
  */
 int8_t z_scouting_config_from(z_owned_scouting_config_t *sc, const z_loaned_config_t *config);
 
 /**
- * Gets the property with the given integer key from the configuration.
+ * Get the property with the given integer key from the configuration.
  *
  * Parameters:
- *   config: A loaned instance of :c:type:`z_owned_scouting_config_t`.
+ *   config: Pointer to a :c:type:`z_loaned_scouting_config_t` to get the property from.
  *   key: Integer key for the requested property.
  *
- * Returns:
- *   Returns the property with the given integer key from the configuration.
+ * Return:
+ *   The requested property value.
  */
 const char *zp_scouting_config_get(const z_loaned_scouting_config_t *config, uint8_t key);
 
 /**
- * Inserts or replaces the property with the given integer key in the configuration.
```

Comment:
`Inserts or replaces` was more relevant, the "update" assumes the key already existed